app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentFragment.kt

```diff
@@ -1,37 +1,58 @@
 package com.sunasterisk.boringweather.ui.current
 
-import android.annotation.SuppressLint
 import android.os.Bundle
-import android.view.LayoutInflater
+import android.view.MenuItem
 import android.view.View
-import android.view.ViewGroup
+import android.widget.Toast
 import androidx.core.view.GravityCompat
 import androidx.core.widget.NestedScrollView
-import androidx.fragment.app.Fragment
+import com.google.android.material.navigation.NavigationView
 import com.sunasterisk.boringweather.R
-import com.sunasterisk.boringweather.util.TemperatureUnit
+import com.sunasterisk.boringweather.base.BaseFragment
+import com.sunasterisk.boringweather.data.CityRepositoryImpl
+import com.sunasterisk.boringweather.data.CurrentRepositoryImpl
+import com.sunasterisk.boringweather.util.UnitSystem
 import com.sunasterisk.boringweather.util.verticalScrollProgress
 import kotlinx.android.synthetic.main.fragment_current.*
+import kotlinx.android.synthetic.main.tile_detail.*
+import kotlinx.android.synthetic.main.tile_summary.*
 
-class CurrentFragment : Fragment() {
+class CurrentFragment : BaseFragment(), CurrentContract.View,
```

Comment:
```suggestion
class CurrentFragment private constructor() : BaseFragment(), CurrentContract.View,
```

Reply:
Bị lỗi khi Apply change and restart activity. System không thể tạo fragment. T chưa thử với khi mà app bị chết cũng bị không.